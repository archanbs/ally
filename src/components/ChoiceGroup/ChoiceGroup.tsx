import React from "react";

export type IChoiceGroupProps = {
  label: string;
  options: IChoiceOption[];
  selectedKey: string;
  onChange: (selected: IChoiceOption) => void;
};

export type IChoiceOption = {
  key: string;
  value: string;
};

export const ChoiceGroup = ({ label, options, selectedKey, onChange }: IChoiceGroupProps) => {
  const name = label.split(" ").join("-");
  return (
    <div>
      <label className="font-semibold my-2">{label}</label>
      <br />
      {options.map(({ key, value }, index) => {
        return (
          <div key={key + index}>
            <input
              type="radio"
              id={key}
              name={name}
              value={value}
              onChange={(ev) => {
                ev.preventDefault();
                console.log("onChange triggered");
                onChange({ key: key, value: value });
              }}
              checked={key === selectedKey}
            />
            <label className="m-2" htmlFor={key}>
              {value}
            </label>
            <br />
          </div>
        );
      })}
      <br />
    </div>
  );
};
