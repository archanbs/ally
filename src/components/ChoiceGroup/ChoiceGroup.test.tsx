import renderer from "react-test-renderer";
import React from "react";
import { ChoiceGroup, IChoiceOption } from "./ChoiceGroup";

test("Choice group snapshot test", () => {
  const options: IChoiceOption[] = [{ key: "1", value: "Choice 1" }];
  const compTree = renderer
    .create(<ChoiceGroup label="Choice group" options={options} onChange={() => console.log("choice change")} selectedKey="1" />)
    .toJSON();
  expect(compTree).toMatchSnapshot();
});
