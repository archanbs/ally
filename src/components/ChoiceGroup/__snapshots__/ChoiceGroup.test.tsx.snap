// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox snapshot test 1`] = `
<div>
  <label
    className="font-semibold my-2"
  >
    Choice group
  </label>
  <br />
  <div>
    <input
      checked={true}
      id="1"
      name="Choice-group"
      onChange={[Function]}
      type="radio"
      value="Choice 1"
    />
    <label
      className="m-2"
      htmlFor="1"
    >
      Choice 1
    </label>
    <br />
  </div>
  <br />
</div>
`;
